feat(rai_cli): improved UX

maciejmajek opened the pull request:
## Purpose

This PR improves user's experience during building documentation vector store and robot identity.

## Proposed Changes

rai_cli now informs the user about possible costs during build.

## Issues



## Testing

- Tested a few times by running 
```bash
poetry run parse_whoami_package src/examples/rosbot-xl-demo/src/rosbot_xl_whoami/description/
```
To recreate testing environment, run:
```bash
vcs import < demos.repos
```

maciejmajek committed:
feat: inform user about the costs, use logging, allow skipping

boczekbartek commented on src/rai/rai/cli/rai_cli.py:
```diff
@@ -35,42 +41,80 @@ def parse_whoami_package():
+            "Your description should be thorough and detailed."
+            "Your reply should start with I am a ..."
+        )
+        llm = ChatOpenAI(model="gpt-4o-mini")
```
Model selection should be added to some global RAI/langchain config - it will become hell for users without OpenAI key to use rai

maciejmajek replied:
Done

boczekbartek commented on src/rai/rai/cli/rai_cli.py:
```diff
@@ -35,42 +41,80 @@ def parse_whoami_package():
+
+    def build_docs_vector_store():
+        logger.info("Building the robot docs vector store...")
+        faiss_index = FAISS.from_documents(docs, OpenAIEmbeddings())
```
Same here, we should avoid hard OpenAI dependency and make it interchangeable with OpenSource solution

maciejmajek replied:
Done

boczekbartek commented on src/rai/rai/cli/rai_cli.py:
```diff
@@ -35,42 +41,80 @@ def parse_whoami_package():
+        f"Building the robot identity will cost "
+        f"approximately {n_tokens / 1_000_000 * 0.15:.4f}$. "
+        "Do you want to continue? (y/n)"
+    )
```
Please mention in the warning that it can be done manually (copy the logger.info message to the warning)

maciejmajek replied:
Done

maciejmajek committed:
feat(config): add config.toml file with LLM and embeddings configurations
feat(model_initialization): add functions to initialize LLM and embeddings models based on config file

maciejmajek committed:
chore: remove cost calculation, enahnce logging

Co-authored-by: Bartłomiej Boczek <bartlomiej.boczek@robotec.ai>

maciejmajek committed:
docs(utils): add Apache License 2.0 header to new __init__.py and model_initialization.py files for legal compliance and clarity

boczekbartek commented on src/rai/rai/utils/model_initialization.py:
```diff
@@ -0,0 +1,73 @@
+
+from typing import Literal, TypedDict
+
+import tomli
```
please update `pyproject.toml`

maciejmajek replied:
Done

boczekbartek commented on src/rai/rai/utils/model_initialization.py:
```diff
@@ -0,0 +1,73 @@
+    vendor: str
+    model: str
+    region_name: str
+    base_url: str
```
Isn't dataclass better for this usecase? Besides accessing the dict/class keys (which comes to personal preference - my is to access `config.vendor` rather than `config["vendor"]`), using dicts might be error prone. Please consider not-full config case:
```
config = EmbeddingsConfig(dict(vendor='abc'))
```
then `model` is missing and will raise KeyError when we try to initialize embeddings.
With dataclass, we ensure that all attributes are set once config is created - so probably at the very beginning of the program

maciejmajek replied:
Makes sense. I've updated the code.

maciejmajek committed:
build: add tomli

maciejmajek committed:
refactor: simplify config, use dataclasses

boczekbartek commented on src/rai/rai/cli/rai_cli.py:
```diff
@@ -35,42 +41,79 @@ def parse_whoami_package():
+
+    logger.info(
+        "Building the robot identity... "
+        "You can do it manually by creating {save_dir}/robot_identity.txt "
```
f-string?
I got this log:
```
Building the robot identity... You can do it manually by creating {save_dir}/robot_identity.txt 
```

```suggestion
        f"You can do it manually by creating {save_dir}/robot_identity.txt "
```

boczekbartek reviewed (approved):
@maciejmajek 1 minor fix. Besides that I've tested the code and it LGTM!

maciejmajek committed:
fix: f string

Co-authored-by: Bartek Boczek <22739059+boczekbartek@users.noreply.github.com>
